tests/test_navigation.py

```diff
@@ -0,0 +1,324 @@
+import unittest
+from unittest.mock import patch
+from webapp.navigation import get_current_page_bubble
+
+
+class TestNavigation(unittest.TestCase):
+    """Test cases for the navigation.py module"""
+
+    def setUp(self):
+        """Set up test data that mirrors the actual
+        secondary-navigation.yaml structure"""
+        self.mock_navigation_data = {
+            "internet-of-things": {
+                "title": "物联网",
+                "path": "/internet-of-things",
+                "children": [
+                    {"title": "Overview", "path": "/internet-of-things"},
+                    {
+                        "title": "Ubuntu Core",
+                        "path": "/internet-of-things/core",
+                    },
+                    {
+                        "title": "智能显示屏",
+                        "path": "/internet-of-things/smart-displays",
+                    },
+                ],
+            },
+            "infrastructure": {
+                "title": "基础架构",
+                "path": "/infrastructure",
+                "children": [
+                    {"title": "Overview", "path": "/infrastructure"},
+                    {"title": "公有云", "path": "/cloud/public-cloud"},
+                    {"title": "OpenStack", "path": "/cloud/openstack"},
+                ],
+            },
+            "desktop": {
+                "title": "桌面系统",
+                "path": "/desktop",
+                "children": [
+                    {"title": "Overview", "path": "/desktop"},
+                    {"title": "特点", "path": "/desktop/features"},
+                ],
+            },
+            "blog": {
+                "title": "博客",
+                "path": "/blog",
+                "children": [{"title": "Overview", "path": "/blog"}],
+            },
+            "no-children": {"title": "No Children", "path": "/no-children"},
+        }
+
+    def _mock_navigation_lookups(self, navigation_data):
+        """Helper method to create mock lookup dictionaries"""
+        bubble_path_lookup = {}
+        child_path_lookup = {}
+        fallback_paths = []
+
+        for bubble_name, bubble_data in navigation_data.items():
+            bubble_path = bubble_data.get("path", "")
+
+            if bubble_path:
+                bubble_path_lookup[bubble_path] = bubble_data
+                fallback_paths.append(bubble_path)
+
+            for child in bubble_data.get("children", []) or []:
+                child_path = child.get("path")
+                if child_path:
+                    child_path_lookup[child_path] = bubble_data
+
+        fallback_paths.sort(key=len, reverse=True)
+
+        return bubble_path_lookup, child_path_lookup, fallback_paths
```

Comment:
Me could have also done with direct lists for bubble_path, child_path and fallback paths.
Just makes the entire logic more simple.